rslib/src/scheduler/bury_and_suspend.rs

```diff
@@ -138,15 +138,26 @@ impl Collection {
 
     pub(crate) fn bury_siblings(
         &mut self,
-        cid: CardId,
+        card: &Card,
         nid: NoteId,
-        bury_mode: BuryMode,
+        mut bury_mode: BuryMode,
     ) -> Result<usize> {
-        let cards = self.storage.all_siblings_for_bury(cid, nid, bury_mode)?;
+        bury_mode.exclude_earlier_gathered_queues(card.queue);
+        let cards = self
+            .storage
+            .all_siblings_for_bury(card.id, nid, bury_mode)?;
         self.bury_or_suspend_cards_inner(cards, BuryOrSuspendMode::BurySched)
     }
 }
 
+impl BuryMode {
+    /// Disables burying for queues gathered before `queue`.
+    fn exclude_earlier_gathered_queues(&mut self, queue: CardQueue) {
+        self.bury_interday_learning &= matches!(queue, CardQueue::DayLearn);
+        self.bury_reviews &= matches!(queue, CardQueue::DayLearn | CardQueue::Review);
```

Comment:
Is this perhaps a bit more restrictive that it needs to be? I think we may only need to enforce this when the user has some burying options enabled but not others. For example, if burying of both new cards and reviews is enabled, it's fine for new cards to bury reviews, since if there's a sibling review card that will take precedence. The issue comes when only some of the burying options are enabled, and eg. a new card ends up burying a review card that we've already gathered.

Reply:
Sorry, my thinking was wrong. Of course, relaxing this restriction if new card burying is enabled would buy us nothing, since the fetch ordering would always ensure that the review sibling takes precedence. Please ignore me. :-)

Reply:
Hmm, thinking a bit more about this, should CardQueue::Learn also be in each of these lines? Intraday learning cards are fetched before anything else, and they should bury siblings (except if they're also intraday learning)

Reply:
Of course, you're right. I missed that one. It also seems like `PreviewRepeat` cards should be taken into account.